src/Illuminate/Console/Concerns/InteractsWithIO.php

```diff
@@ -245,10 +245,10 @@ public function table($headers, $rows, $tableStyle = 'default', array $columnSty
      * @param  \Closure  $callback
      * @return mixed|void
      */
-    public function withProgressBar($totalSteps = 0, Closure $callback)
+    public function withProgressBar($totalSteps, Closure $callback)
     {
         $bar = $this->output->createProgressBar(
-            is_iterable($totalSteps) ? count($totalSteps) : $totalSteps
+            is_iterable($totalSteps) ? count($totalSteps) : (is_int($totalSteps) ? $totalSteps : 0)
```

Comment:
`is_int` is always true in that scenario. Also, iterables aren't always countable, so this code is bad anyway...